arm-keyvault/2015-06-01/swagger/keyvault.json

```diff
@@ -0,0 +1,463 @@
+        "name": {
+          "type": "string",
+          "description": "SKU name"
+        }
+      },
+      "required": [
+        "family",
+        "name"
+      ]
+    },
+    "AccessPolicyEntry": {
+      "properties": {
+        "tenantId": {
+          "type": "string",
+          "format": "uuid",
+          "description": "Tenant ID of the principal"
+        },
+        "objectId": {
+          "type": "string",
+          "format": "uuid",
+          "description": "Object ID of the principal"
+        },
+        "applicationId": {
+          "type": "string",
+          "format": "uuid",
+          "description": " Application ID of the client making request on behalf of a principal"
+        },
+        "permissions": {
+          "$ref": "#/definitions/Permissions",
+          "description": "Permissions that the principal has for this vault"
+        }
+      }
+    },
+    "Permissions": {
+      "properties": {
+        "keys": {
+          "type": "array",
+          "items": {
+            "type": "string"
+          },
+          "description": "Permissions to keys"
+        },
+        "secrets": {
+          "type": "array",
+          "items": {
+            "type": "string"
+          },
+          "description": "Permissions to secrets"
+        }
+      }
+    },
+    "VaultProperties": {
+      "properties": {
+        "vaultUri": {
+          "type": "string",
+          "description": "URL of the vault"
+        },
+        "tenantId": {
+          "type": "string",
+          "format": "uuid",
+          "description": "Tenant ID"
+        },
+        "sku": {
+          "$ref": "#/definitions/Sku",
+          "description": "SKU details"
+        },
+        "accessPolicies": {
+          "type": "array",
+          "items": {
+            "$ref": "#/definitions/AccessPolicyEntry"
+          },
+          "description": "Access policies for one or more principals"
+        },
+        "enabledForDeployment": {
+          "type": "boolean",
+          "description": "Enabled or disabled for deployment"
+        },
+        "enabledForDiskEncryption": {
+          "type": "boolean",
+          "description": "Enabled or disabled for disk encryption"
+        },
+        "enabledForTemplateDeployment": {
+          "type": "boolean",
+          "description": "Enabled or disabled for Azure Resource Manager template deployment"
+        }
+      },
+      "required": [
+        "tenantId",
+        "sku",
+        "accessPolicies"
+      ],
+      "description": "Properties of the vault"
+    },
+    "VaultCreateOrUpdateParameters": {
+      "properties": {
+        "location": {
+          "type": "string",
+          "description": "Resource location"
+        },
+        "tags": {
+          "type": "object",
+          "additionalProperties": {
+            "type": "string"
+          },
+          "description": "Resource tags"
+        },
+        "properties": {
+          "x-ms-client-flatten": false,
```

Comment:
If you do not want flattening then it is fine. You can remove it. AutoRest will not flatten it if the extension is not present. Setting it to false and not having the extension both mean the same :)